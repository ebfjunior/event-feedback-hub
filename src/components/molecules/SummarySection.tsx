import React from 'react';
import { BulletList } from '@/components/atoms/BulletList';

export type SummarySectionProps = {
  icon: React.ReactNode;
  title: string;
  items?: string[] | null;
};

export function SummarySection({ icon, title, items }: SummarySectionProps) {
  return (
    <section className="rounded-md bg-muted/40 p-3">
      <div className="mb-2 flex items-center gap-2 text-sm font-semibold">
        {icon}
        <span>{title}</span>
      </div>
      <BulletList className="list-inside space-y-2 text-sm text-muted-foreground" items={items ?? []} />
    </section>
  );
}


