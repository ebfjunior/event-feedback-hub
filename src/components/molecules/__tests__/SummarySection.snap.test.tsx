import React from 'react';
import { render } from '@testing-library/react';
import { SummarySection } from '../SummarySection';

describe('SummarySection (snapshot)', () => {
	it('renders with items', () => {
		const { container } = render(
			<SummarySection icon={<span>i</span>} title="Highlights" items={["Nice", "Great"]} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders with empty items', () => {
		const { container } = render(
			<SummarySection icon={<span>i</span>} title="Highlights" items={[]} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});


