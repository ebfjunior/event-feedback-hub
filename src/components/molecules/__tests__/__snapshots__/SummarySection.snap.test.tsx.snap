// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SummarySection (snapshot) > renders with empty items 1`] = `
<section
  class="rounded-md bg-muted/40 p-3"
>
  <div
    class="mb-2 flex items-center gap-2 text-sm font-semibold"
  >
    <span>
      i
    </span>
    <span>
      Highlights
    </span>
  </div>
  <ul
    class="list-inside space-y-2 text-sm text-muted-foreground"
  >
    <li
      class="flex gap-2"
    >
      <span
        class="select-none"
      >
        •
      </span>
      <span>
        —
      </span>
    </li>
  </ul>
</section>
`;

exports[`SummarySection (snapshot) > renders with items 1`] = `
<section
  class="rounded-md bg-muted/40 p-3"
>
  <div
    class="mb-2 flex items-center gap-2 text-sm font-semibold"
  >
    <span>
      i
    </span>
    <span>
      Highlights
    </span>
  </div>
  <ul
    class="list-inside space-y-2 text-sm text-muted-foreground"
  >
    <li
      class="flex gap-2"
    >
      <span
        class="select-none"
      >
        •
      </span>
      <span>
        Nice
      </span>
    </li>
    <li
      class="flex gap-2"
    >
      <span
        class="select-none"
      >
        •
      </span>
      <span>
        Great
      </span>
    </li>
  </ul>
</section>
`;
